compose/final/app/src/main/java/com/apollographql/apollo3/extensions/compose/paging/ApolloPagingSource.kt

```diff
@@ -0,0 +1,96 @@
+package com.apollographql.apollo3.extensions.compose.paging
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import androidx.paging.compose.LazyPagingItems
+import androidx.paging.compose.collectAsLazyPagingItems
+import com.apollographql.apollo3.ApolloCall
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.Operation
+import com.apollographql.apollo3.extensions.compose.exception
+import com.apollographql.apollo3.extensions.compose.tryExecute
+
+class ApolloPagingSource<Data : Operation.Data, Value : Any>(
+    /**
+     * The call to fetch the next page, given the response for the current page.
+     * This is also used for the initial/refresh call.
+     * The response will be null for the initial/refresh call.
+     */
+    private val appendCall: (response: ApolloResponse<Data>?) -> ApolloCall<Data>,
+
+    /**
+     * Whether there is a next page, given the response for the current page.
+     */
+    private val hasNextPage: (response: ApolloResponse<Data>) -> Boolean,
+
+    /**
+     * The call to fetch the previous page, given the response for the current page.
+     * Can be null if prepend is not supported.
+     */
+    private val prependCall: ((response: ApolloResponse<Data>) -> ApolloCall<Data>)? = null,
+
+    /**
+     * Whether there is a previous page, given the response for the current page.
+     */
+    private val hasPreviousPage: (response: ApolloResponse<Data>) -> Boolean = { false },
+
+    /**
+     * Extract the list of items from a response.
+     * Throw an exception if the response cannot be used (e.g. it contains GraphQL errors).
```

Comment:
Could be a `Result<List<Value>>` instead.
- Forces implementor to not forget the error case
- A bit of friction if you _don't_ have GraphQL error cases